lib/tls.js

```diff
@@ -143,9 +146,13 @@ exports.convertALPNProtocols = function convertALPNProtocols(protocols, out) {
   // If protocols is Array - translate it into buffer
   if (ArrayIsArray(protocols)) {
     out.ALPNProtocols = convertProtocols(protocols);
-  } else if (isArrayBufferView(protocols)) {
+  } else if (Buffer.isBuffer(protocols) || isUint8Array(protocols)) {
     // Copy new buffer not to be modified by user.
```

Comment:
I think it is to handle cases where the user mutates the input buffer after `out.ALPNProtocols` is assigned? Not sure.

Reply:
Yep, it's because user keeps reference to the buffer and can mutate it afterwards. Copying was introduced by https://github.com/nodejs/node/pull/8055.